dipy/nn/synb0.py

```diff
@@ -0,0 +1,417 @@
+        except ValueError:
+            raise ValueError('Expected input for the provided model weights \
+                             do not match the declared model')
+
+    def __predict(self, x_test):
+        r"""
+        Internal prediction function
+
+        Parameters
+        ----------
+        x_test : np.ndarray (batch, 80, 80, 96, 2)
+            Image should match the required shape of the model.
+
+        Returns
+        -------
+        np.ndarray (...) or (batch, ...)
+            Reconstructed b-inf image(s)
+        """
+
+        return self.model.predict(x_test)
+
+    def __normalize(self, image, max_img, min_img):
+        r"""
+        Internal normalization function
+
+        Parameters
+        ----------
+        image : np.ndarray
+
+        Returns
+        -------
+        np.ndarray
+            Normalized image from range -1 to 1
+        """
+        return ((image - min_img)/(max_img - min_img))*2-1
+
+    def __unnormalize(self, image, max_img, min_img):
+        r"""
+        Internal unnormalization function
+
+        Parameters
+        ----------
+        image : np.ndarray
+
+        Returns
+        -------
+        np.ndarray
+            unnormalized image from range min_img to max_img
+        """
+        return (image+1)/2*(max_img-min_img) + min_img
+
+    def predict(self, b0, T1, batch_size=None, average=True):
+        r"""
+        Wrapper function to faciliate prediction of larger dataset.
+        The function will scale the data to meet the required shape of image.
+        Note that the b0 and T1 image should have the same shape
+
+        Parameters
+        ----------
+        b0 : np.ndarray (batch, 77, 91, 77) or (77, 91, 77)
+            For a single image, input should be a 3D array. If multiple images,
+            there should also be a batch dimension.
+
+        T1 : np.ndarray (batch, 77, 91, 77) or (77, 91, 77)
+            For a single image, input should be a 3D array. If multiple images,
+            there should also be a batch dimension.
+
+        batch_size : int
+            Number of images per prediction pass. Only available if data
+            is provided with a batch dimension.
+            Consider lowering it if you get an out of memory error.
+            Increase it if you want it to be faster and have a lot of data.
+            If None, batch_size will be set to 1 if the provided image
+            has a batch dimension.
+            Default is None
+        Returns
+        -------
+        pred_output : np.ndarray (...) or (batch, ...)
+            Reconstructed b-inf image(s)
+
+        """
+        if all([b0.shape[1:] != (77, 91, 77), b0.shape != (77, 91, 77)]) or \
+                b0.shape != T1.shape:
+            raise ValueError('Expected shape (batch, 77, 91, 77) or \
+                             (77, 91, 77) for both inputs')
+
+        shape = b0.shape
+        dim = len(shape)
+
+        if dim == 3:
```

Comment:
Some documentation about this pre-processing would be useful.

Reply:
Good idea. Will add it on the next commit.